### ISSUE TITLE
KATANA- wrong versions for estimate fee

### ISSUE THREAD

**Issue by PhilippeR26** [dojoengine/dojo#806]
**Describe the bug**
It has been reported that Starknet.js do not work with Katana.
I analyzed this problem. The transaction fails only for `estimate_fee`.
Starknet.js is conform to Cairo-lang : 
https://github.com/starkware-libs/cairo-lang/blob/d6863bdebf2e71f90a64e0c5f6b021b0d07b61cf/src/starkware/starknet/definitions/constants.py#L63-L71
It uses for estimate version : 2^128+2
Katana isn't conform to Cairo-lang, and use estimate version : 2
This difference generates a different message_Hash, and the validation of the signature fails.

**To Reproduce**
```bash
katana --accounts 3 --seed 0 --port 5001
```

TypeScript :
```typescript
const provider = new RpcProvider({ nodeUrl: 'http://0.0.0.0:5001' });
const privateKey = "0x300001800000000300000180000000000030000000000003006001800006600";
const accountAddress = "0x3ee9e18edc71a6df30ac3aca2e0b02a198fbce19b7480a63a0d71cbd76652e0";
const account0 = new Account(provider, accountAddress, privateKey, "0");
const declareResponse = await account0.declare({ contract: compiledSierra, casm: compiledCasm });
```
As no maxFee has been provided, Starknet.js  performs an estimateFee ... that fail.
If maxFee is provided, Starknet.js do not launch an estimateFee ... and the command succeed :
```typescript
const declareResponse = await account0.declare({ contract: compiledSierra, casm: compiledCasm }, { maxFee: 9_000_000_000_000_000 });
```
**Expected behavior**
Katana should use the same estimate versions than Cairo_lang.

**Screenshots**

**Additional context**

**Comment by kariy** [comment]
seems to be the same issue as #792 

are you on the latest commit? should be fixed in #793

**Comment by PhilippeR26** [comment]
commit 374512a71f452e3b7982d660bb54ea5e6ef4f786 of 18/aug/2023 (one day before this issue creation).
